pkg/reconciler/testing/v1/subscription.go

```diff
@@ -221,12 +221,12 @@ func WithSubscriptionPhysicalSubscriptionReply(reply *duckv1.Addressable) Subscr
 	}
 }
 
-func WithSubscriptionDeadLetterSinkURI(uri *apis.URL) SubscriptionOption {
+func WithSubscriptionDeadLetterSinkURI(uri *duckv1.Addressable) SubscriptionOption {
```

Comment:
Shouldn't we rename the function (and its parameters), as we don't take an URI anymore?